orb-ui/cone/examples/cone-simulation.rs

```diff
@@ -0,0 +1,188 @@
+use color_eyre::eyre::{eyre, Context};
+use tokio::sync::broadcast;
+use tokio::sync::broadcast::error::RecvError;
+use tokio::task;
+use tokio::task::JoinHandle;
+use tracing::level_filters::LevelFilter;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::{fmt, EnvFilter};
+
+use orb_cone::led::CONE_LED_COUNT;
+use orb_cone::{ButtonState, ConeEvent};
+use orb_rgb::Argb;
+
+const CONE_LED_STRIP_DIMMING_DEFAULT: u8 = 10_u8;
+const CONE_SIMULATION_UPDATE_PERIOD_S: u64 = 2;
+const CONE_LED_STRIP_MAXIMUM_BRIGHTNESS: u8 = 20;
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    let registry = tracing_subscriber::registry();
+    #[cfg(tokio_unstable)]
+    let registry = registry.with(console_subscriber::spawn());
+    registry
+        .with(fmt::layer())
+        .with(
+            EnvFilter::builder()
+                .with_default_directive(LevelFilter::INFO.into())
+                .from_env_lossy(),
+        )
+        .init();
+
+    let devices = ftdi_embedded_hal::libftd2xx::list_devices()?;
+    for device in devices.iter() {
+        tracing::debug!("Device: {:?}", device);
+    }
+
+    loop {
+        let (tx, mut rx) = broadcast::channel(10);
+        if let Ok((mut cone, cone_handles)) = orb_cone::Cone::spawn(tx) {
+            // spawn a thread to receive events
+            let button_listener_task: JoinHandle<eyre::Result<()>> =
+                task::spawn(async move {
+                    let mut button_state = ButtonState::Released;
+                    loop {
+                        match rx.recv().await {
+                            Ok(event) => match event {
+                                ConeEvent::Button(state) => {
+                                    if state != button_state {
+                                        tracing::info!("🔘 Button {:?}", state);
+                                        button_state = state;
+                                    }
+                                }
+                                ConeEvent::Cone(state) => {
+                                    tracing::info!("🔌 Cone {:?}", state);
+                                }
+                            },
+                            Err(RecvError::Closed) => {
+                                return Err(eyre!(
+                                    "Cone events channel closed, cone disconnected?"
+                                ))
+                            }
+                            Err(RecvError::Lagged(skipped)) => {
+                                tracing::warn!("🚨 Skipped {} cone events", skipped);
+                            }
+                        }
+                    }
+                });
+
+            // create one shot to gracefully terminate simulation
+            let (kill_sim_tx, mut kill_sim_rx) = tokio::sync::oneshot::channel::<()>();
+            let simulation_task: JoinHandle<eyre::Result<()>> = tokio::task::spawn(
```

Comment:
nit: simulation task could just be an async fn instead of a task. This would simplify things because you no longer need to have kill signals or think about the tasks lifecycle independently from the main function's future.

I don't have concrete guidance on when to make something a task vs just a regular future. But I *think* that here, a future makes more sense.